### PR TITLE
Eureka url can not be set from xml config #36

### PR DESCRIPTION
**Pull request by omidp**
Fixes: https://github.com/hazelcast/hazelcast-eureka/issues/36

### CODE REVIEW

**Comment by hasancelik** [comment]
fixes https://github.com/hazelcast/hazelcast-eureka/issues/36

**Comment by hasancelik** [comment]
run-lab-run

**Comment by hasancelik** [comment]
Thanks for the PR @omidp 👍 

PR builder job failed due to checkstyle errors, you can check them with:
```
mvn clean install -DskipTests -Pcheckstyle,spotbugs
```
Could you fix them and update the PR?

File: hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java [hunk 1]
```diff
@@ -1332,18 +1332,25 @@ private void handleAliasedDiscoveryStrategy(WanBatchPublisherConfig publisherCon
...
             if ("enabled".equals(lowerCaseInternal(att.getNodeName()))) {
                 config.setEnabled(getBooleanValue(value));
             } else if (att.getNodeName().equals("connection-timeout-seconds")) {
                 config.setProperty("connection-timeout-seconds", value);
             }
         }
+        LowercaseHandler lowerCaseHandler = new LowercaseHandler();
         for (Node n : childElements(node)) {
+        	lowerCaseHandler.put(n.getLocalName(), "eureka".equals(n.getParentNode().getLocalName()));
             String key = cleanNodeName(n);
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java]: Can't we simply pass the `tag` to this metrod from the caller and do the following without the need for the handler?
```
String key = cleanNodeName(n, !tag.equals("eureka"));

[...]

public static String cleanNodeName(final Node node, final boolean shouldLowerCase) {
[...]
}
```
This seems we put a "local" boolean to a hashmap that we lookup for a few lines later in the same cycle, so the handler and the map feel unnecessary here.

File: hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java [hunk 2]
```diff
@@ -407,6 +407,8 @@ public void readEurekaConfig() {
                 + "            <eureka enabled=\"true\">\n"
                 + "                <use-public-ip>true</use-public-ip>\n"
                 + "                <namespace>hazelcast</namespace>\n"
+                + "                <shouldUseDns>false</shouldUseDns>\n"
+                + "				   <serviceUrl.default>http://localhost:8082/eureka</serviceUrl.default>\n"
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java]: Could you please add an assertion for the two added properties and do the same changes in the `YamlConfigBuilderTest` too?

**Comment by omidp** [comment]
> Thanks for the PR @omidp 
> 
> PR builder job failed due to checkstyle errors, you can check them with:
> 
> ```
> mvn clean install -DskipTests -Pcheckstyle,spotbugs
> ```
> 
> Could you fix them and update the PR?

Done. I had it fixed. 
Does Hazelcast have any Eclipse Java code formatter ? currently, I can not use IDE code formatter

File: hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java (hunk 2 above)
Comment (omidp) [reply hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java]: Done

File: hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java (hunk 1 above)
Comment (omidp) [reply hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java]: cleanNodeName is being used in a lot of places I wanted to have less impact in the codebase

Comment (blazember) [reply hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java]: I see your point @omidp. You can keep the change local with overriding the `cleanNodeName` method like
```
    public static String cleanNodeName(final Node node) {
        return cleanNodeName(node, true);
    }

    public static String cleanNodeName(final Node node, final boolean shouldLowerCase) {
        final String nodeName = node.getLocalName();
        if (nodeName == null) {
            throw new HazelcastException("Local node name is null for " + node);
        }
        return shouldLowerCase ? StringUtil.lowerCaseInternal(nodeName) : nodeName;
    }
```
So no other code needs to be modified and we follow the common logic for eureka config processing as well. I'd prefer this over introducing a helper class and a map.

Comment (omidp) [reply hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java]: Ok done

**Comment by Holmistr** [comment]
@omidp No, I don't think we have an Eclipse Java code formatter. Perhaps @kwart knows?

**Comment by kwart** [comment]
Here are my eclipse configs:
[hazelcast-eclipse-configs.zip](https://github.com/hazelcast/hazelcast/files/4254306/hazelcast-eclipse-configs.zip)

File: hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java [hunk 3]
```diff
@@ -419,6 +421,7 @@ public void readEurekaConfig() {
         assertTrue(eurekaConfig.isEnabled());
         assertTrue(eurekaConfig.isUsePublicIp());
         assertEquals("hazelcast", eurekaConfig.getProperty("namespace"));
+        assertEquals("false", eurekaConfig.getProperty("shouldUseDns"));
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java]: Could you please add assertion for `serviceUrl.default` too?

File: hazelcast/src/main/java/com/hazelcast/internal/config/DomConfigHelper.java [hunk 4]
```diff
@@ -98,6 +98,14 @@ public static String cleanNodeName(final Node node) {
         }
         return StringUtil.lowerCaseInternal(nodeName);
     }
+    
+    public static String cleanNodeName(final Node node, final boolean shouldLowercase) {
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/internal/config/DomConfigHelper.java]: You can call this method from `cleanNodeName(Node)` with `true`, so the duplication can be removed.

**Comment by blazember** [comment]
run-lab-run

**Comment by blazember** [comment]
@omidp Thank you for addressing my comments that quickly. There are two trailing white spaces that breaks the checkstyle rules. Could you please check and fix it in local with 
```
mvn clean install -DskipTests -Pcheckstyle,spotbugs
```

**Comment by omidp** [comment]
@blazember  please check it should be ok now.

**Comment by blazember** [comment]
run-lab-run

**Comment by Holmistr** [comment]
@omidp We're almost done! Could I ask you to sign our Contributor's agreement?
https://hazelcast.atlassian.net/wiki/spaces/COM/pages/6357071/Hazelcast+Contributor+Agreement

**Comment by Holmistr** [comment]
Hi @omidp , is everything OK? We're very close to merging your PR, we just need the CLA signed :) I know, it's a horrible process, we're really working on a better solution (one click on GitHub). We're looking forward to your contribution very much :)

**Comment by omidp** [comment]
> Hi @omidp , is everything OK? We're very close to merging your PR, we just need the CLA signed :) I know, it's a horrible process, we're really working on a better solution (one click on GitHub). We're looking forward to your contribution very much :)


Hi @Holmistr I just sent the CLA. Thank you for your guideline.

**Comment by Holmistr** [comment]
@omidp CLA received, merging the PR! :tada:  Thank you very much for your contribution!

**Pull request merged**